### ISSUE TITLE
Indent lint regression in 3.12.2

### ISSUE THREAD

**Issue by jfirebaugh** [eslint/eslint#7771]
**test.js**:

```js
console.log(
    true ?
        "a" :
        ["b",
            []]
);
```

Output of `eslint --no-eslintrc --rule='indent: 2' test.js`:

```
  4:9  error  Expected indentation of 4 spaces but found 8  indent
```

An indent of 4 spaces does not make sense for line 4. That would look like:

```js
console.log(
    true ?
        "a" :
    ["b",
            []]
);
```

Another indication that this is a bug is that the following produces no errors:

```js
console.log(
    true ?
        "a" :
        ["b", []]
);
```

Furthermore, version 3.12.1 has no issues with this example.

Looking in the 3.12.2 changes, it seems that #7747 / #7750 may be related.

**Comment by not-an-aardvark** [comment]
I was able to reproduce this. Thanks for the report.

**Comment by alexindigo** [comment]
Another regression with function expressions:

code: ![code](http://ia.gs/452P1y210q2D/Image%202016-12-18%20at%2012.15.45%20AM.png)

error:

```
 148:5  error  Expected indentation of 6 spaces but found 4  indent
```

If I wrap function in brackets, it doesn't error.
Previous version of eslint parsed it fine.

**Comment by not-an-aardvark** [comment]
@alexindigo That looks like intended behavior -- your function is in an array, so it should be indented from the start of the array. Based on inferring what the code around your sample looks like, the rule correctly reports an error in both 3.11.1 and 3.11.2.

**Comment by alexindigo** [comment]
@not-an-aardvark So previous element is indented with 4 spaces and it's ok, but the next element indented with 4 spaces is not ok, just because it's a function? Or did I miss anything?

Example array:

```
[
  25,

  'abc',

  ['sub array'],

    function extraIndented() { return 'Why Oh Why?'; },

  someVariable,

  (function itsOkWithInBrackets() { return 'totaly not confusing'; })
]
```

:)

**Comment by not-an-aardvark** [comment]
You have an extra opening bracket on line 145 of your original example, so the `function(cb)` expression is actually an element of the array that starts on that line. With the example array in your most recent comment, the rule expects the function to have an indentation of 2 spaces.

**Comment by alexindigo** [comment]
Ha :)) You're so right. I feel dumb now. :) Interesting it didn't caught it before.
Thanks for the patient explanation. 👍

**Comment by mourner** [comment]
Confirmed with `git bisect` that #7750 is indeed the culprit. @not-an-aardvark
